### PR TITLE
25 search users

### PR DESCRIPTION
**Pull request by Amoodaa**
SORRY FOR the 1 commit :d
i renamed some files for consistency
relates #25

### CODE REVIEW

File: server/controllers/users/searchUsers.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { users: { searchUsers } } = require('../../database/queries');
+
+exports.searchUsers = (req, res, next) => {
+  // we will use req.query to get results based on it
+  const { search: searchQuery } = req.query;
+  searchUsers(searchQuery)
+    .then(({ rows }) => {
+      if (rows.length && rows) {
```
Comment (11janci) [inline server/controllers/users/searchUsers.js]: pls turn the condition around to `if (rows && rows.length)` (if rows.length is true that means rows is automatically true as well and no need to check it, however if rows is null or undefined, you will get an error saying `property length does not exist for undefined` or sth like that

File: server/database/queries/users/searchUsers.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 const dbConnection = require('../../config/dbConnection');
 
-exports.searchUsers = () => {
+exports.searchUsers = (query) => {
   const sql = {
-    text: '',
-    values: [],
+    text: 'SELECT * FROM users WHERE username LIKE $1 ',
```
Comment (11janci) [inline server/database/queries/users/searchUsers.js]: btw, if I'm not mistaken, the users table stores the passwords correct? Selecting every column and then passing the data to the response will send all of the selected users' passwords to the client! That's a pretty big security flaw (even thogh the passwords are hashed, you most certainly do not want to expose them :) 
This is why it's generally good to store passwords in a separate table (even better, use 3rd party authentication providers). As a quick fix, you can simply exclude the password column from the query (this should be done for all queries from the user table..)

File: server/controllers/users/searchUsers.js (hunk 1 above)
Comment (Amoodaa) [reply server/controllers/users/searchUsers.js]: oh yes good note

File: server/database/queries/users/searchUsers.js (hunk 2 above)
Comment (Amoodaa) [reply server/database/queries/users/searchUsers.js]: WAIT UR RIGHT LMAO
BIG THANKS

File: server/controllers/users/searchUsers.js (hunk 1 above)
Comment (Fatmasiam) [inline server/controllers/users/searchUsers.js]:
```suggestion
      if ( rows[0] ) {
```

Comment (11janci) [reply server/controllers/users/searchUsers.js]: if rows is null or undefined, `rows[0]` will throw an error. You might argue that the searchUsers method never returns undefined or null, but that might change in the future (for example if you upgrade the underlying dependency for querying the DB). It is better to program defensively, I would keep the check (just turn it around as mentioned before)

Comment (Fatmasiam) [reply server/controllers/users/searchUsers.js]: @11janci. oh thx  to notify me for this point

File: server/database/queries/users/searchUsers.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 const dbConnection = require('../../config/dbConnection');
 
-exports.searchUsers = () => {
+exports.searchUsers = (query) => {
   const sql = {
-    text: '',
-    values: [],
+    text: 'SELECT users.id, users.username, email, bio, avatar_path, users.native_lang_id, users.learning_lang_id, COALESCE(json_agg(interests) FILTER(WHERE interests.id IS NOT NULL),\'[]\') AS interests FROM users LEFT OUTER JOIN user_interest ON user_interest.user_id = users.id JOIN interests ON interests.id = user_interest.interest_id WHERE users.isactive = true AND users.username LIKE $1 GROUP BY users.id LIMIT 100',
+    values: [`%${query}%`],
```
Comment (fadeomar) [inline server/database/queries/users/searchUsers.js]: values: [`%${query}%`] 
what it does mean ?

Comment (fadeomar) [reply server/database/queries/users/searchUsers.js]: and please explain the ...rest in the formatlangV2

Comment (Amoodaa) [reply server/database/queries/users/searchUsers.js]: its called the rest operator
check it here https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/rest_parameters
Rest parameters - JavaScript | MDN

Comment (Amoodaa) [reply server/database/queries/users/searchUsers.js]: the query is hard to explain in text
sql like operator takes something like a regex
and here im inserting the text in the regex that matches the inserted query

'amo' doesnt match 'amoodaa'
'%amo' doesnt match 'amoodaa'
'amo%' matches amoodaa but doesnt match 'mamoodaa'
'%amo%' matches amoodaa and matches 'mamoodaa'

**Pull request merged**